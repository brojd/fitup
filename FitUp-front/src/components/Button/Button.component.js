import React, { PropTypes } from 'react';
import classNames from 'classnames';

const Button = (props) => ((
  <button className={classNames('uk-button', props.className)}
          type={props.type}
          onClick={props.onClick}
          value={props.value}
          disabled={props.disabled}
          autoFocus={props.autoFocus} >
    {props.text}
  </button>
));

Button.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func,
  value: PropTypes.string,
  disabled: PropTypes.bool,
  autoFocus: PropTypes.bool
};

Button.defaultProps = {
  type: 'button',
  disabled: false,
  autoFocus: false
};

export default Button;
