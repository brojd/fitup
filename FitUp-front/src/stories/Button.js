import React from 'react';
import { storiesOf} from '@kadira/storybook';
import Button from '../components/Button/Button.component';

storiesOf('Button', module)
  .add('enable advanced search', () => (
    <Button text="Advanced search" type="button" className="uk-button-mini uk-border-rounded my_button" />
  ))
  .add('search events', () => (
    <Button text="Search" type="button" className="uk-button-small uk-button-success uk-border-rounded" />
  ));
