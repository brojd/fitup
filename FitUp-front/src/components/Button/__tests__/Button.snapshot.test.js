import React from 'react';
import Button from '../Button.component';
import renderer from 'react-test-renderer';

test('Button renders correctly with given props', () => {
  const someFunc = jest.fn();
  const component = renderer.create(
    <Button text='Press'
            className='Button'
            onClick={someFunc}
            disabled={false}
            autoFocus={false} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
