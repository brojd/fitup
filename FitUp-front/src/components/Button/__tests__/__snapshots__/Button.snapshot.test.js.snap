exports[`test Button renders correctly with given props 1`] = `
<button
  autoFocus={false}
  className="uk-button Button"
  disabled={false}
  onClick={[Function]}
  type="button"
  value={undefined}>
  Press
</button>
`;
